### DISCUSSION TITLE
Uncover and httpx

### DISCUSSION THREAD

**Discussion by HachimanSec** [projectdiscovery/uncover#314] [Q&A] [answered]
Hello,
I try to run uncover with httpx, and later with nuclei. However it doesnt really work.

What I do is:
`uncover -q TARGET -v -f host | httpx -sc -title -server`

I do see the output of uncover, but nothing from httpx:
```
uncover -q TARGET -v -f host | httpx -sc -title -server -cname -v

  __  ______  _________ _   _____  _____
 / / / / __ \/ ___/ __ \ | / / _ \/ ___/
/ /_/ / / / / /__/ /_/ / |/ /  __/ /
\__,_/_/ /_/\___/\____/|___/\___/_/

                projectdiscovery.io


    __    __  __       _  __
   / /_  / /_/ /_____ | |/ /
  / __ \/ __/ __/ __ \|   /
 / / / / /_/ /_/ /_/ /   |
/_/ /_/\__/\__/ .___/_/|_|
             /_/

                projectdiscovery.io

[INF] Current uncover version v1.0.6 (latest)
[INF] Current httpx version v1.3.5 (latest)
[shodan] TARGET.de
```

If I take one result from uncover and manually run httpx, it works. For example:
```
httpx -sc -title -server -cname -u nas.TARGET.de

    __    __  __       _  __
   / /_  / /_/ /_____ | |/ /
  / __ \/ __/ __/ __ \|   /
 / / / / /_/ /_/ /_/ /   |
/_/ /_/\__/\__/ .___/_/|_|
             /_/

                projectdiscovery.io

[INF] Current httpx version v1.3.5 (latest)
https://nas.TARGET.de [200] [TARGET GL] [nginx]
```

What is it that I am missing here? I would expect piping uncover to nuclei produces similiar results like manually running httpx with the host name.

Thx!

**Comment by ehsandeep** [comment, marked as answer]
@HachimanSec you don't need to use `-v` for using it, it's for debugging purposes, and when used it's outputs data on stderr which will not passed to next tool in the pipeline!

Just remove the `-v` option and everything should work as expected.

**Reply by HachimanSec** [reply]
Thx @ehsandeep ! This really did the trick... I guess I got stuck in a rabbit hole.

**Reply by HachimanSec** [reply]
Sorry, one more question. Is it possible to add additional args to https, save the output to a file and still pipe it nuclei afterwards?
Like this:
`echo TARGET | uncover | httpx -ss -fr -srd -sc -title -server -cname ./httpx | nuclei -o ./nuclei.txt `

**Reply by MetzinAround** [reply]
If you need to keep the httpx results and use them in nuclei, but need a record of the results, I think doing the pipe through httpx and outputting to a file first and then running a separate scan after using the txt file from the httpx scan. 

`echo TARGET | uncover | httpx -ss -fr -srd -sc -title -server -cname ./httpx -o httpxresults.txt`
`nuclei -l httpxresults.txt -u TARGET -o ./nuclei.txt`

**Reply by MetzinAround** [reply]
you have two txt files then, in case you need them both.

**Reply by matinkunduzi511-star** [reply]
Hello, excuse me, can you unblock my WhatsApp, please?

**Comment by matinkunduzi511-star** [comment]
Hello, excuse me, can you unblock my WhatsApp, please?
